src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -1976,8 +1982,8 @@ public synchronized void run() {
         private final Executor mExecutor = Executors.newSingleThreadExecutor();
     }
 
-    private class UnsupportedUpdatesFromMixpanel implements UpdatesFromMixpanel {
-        public UnsupportedUpdatesFromMixpanel(Tweaks tweaks) {
+    /* package */ class NoOpUpdatesFromMixpanel implements UpdatesFromMixpanel {
```

Comment:
"UnsupportedUpdatesFromMixpanel" doesn't describe what this class does. It describes one use case of this class (i.e. when the API version doesn't support viewcrawler ). 

"NoOpUpdatesFromMixpanel" describes what is does ( i.e. nothing, "NoOp" )

It may, in fact, be supported by the API of the device, but just disabled by config, so I feel "Unsupported" is confusing in that case.